H2JC/H2JC
# Redux Persist Implementation

**Major Changes:**

- I added Redux Persist which allows users to refresh and it'll pick up where they left off. 
- I went on every page that I remembered and fixed the rehydration issue that nextjs was giving. The problem is we can't load components until the state is initialized which nextjs doesn't like so the way to fix it is to conditionally render the component only when it is mounted.
- Changing userInfoReducer to signUpReducer for better naming convention
- Changed on MUI buttons in initialSignup components to reg buttons for rehydration issues. 
- Added a persist purge to cover all ends when user signs up as a new user. It is in the log out and signup buttons throughout the application

**Minor Changes:**
- "x" on interests modal does not save when clicked on now. 
- Added reselect so it memorizes the state and reduces rerendering
- Changed all buttons in the initial Signup components to use global styling.


**How to test:**
- Go through the application and reload the page as much as possible, you'll notice that it does take a while to load sometimes but that's expected. We need to do clean up on some components or add a skeleton loading components. 
- Move around the application, like create a user then navigate out and create another before submitting creator data. just do some weird testing and make sure edge cases are cover.
- Go to setting page and notification, notice update button is disable until changes are made. 

REFRESH REFRESH REFRESH
